src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -73,6 +77,10 @@ pub enum AsyncStorageWorkerResponse<T: Timestamp + Lattice + Codec64> {
         /// have already been durably ingested.
         source_resume_uppers: BTreeMap<GlobalId, Vec<Row>>,
     },
+
+    /// Successful processing of Dataflow being dropped.  For AsyncStorageWorker,
+    /// this is currently a no-op.
+    DataflowDropped(GlobalId),
```

Comment:
I know this matches `FrontiersUpdated`, but I'd prefer `DropDataflow` for this. Largely because `DroppedDataflow` sounds like the dataflow has already been dropped while this is really a forward command to drop the dataflow.

Also don't want to be pedantic, though, so if you want to save a CI cycle you can merge as is.